File: src/main/java/com/pairingplanet/pairing_planet/dto/recipe/TrendingTreeDto.java
package com.pairingplanet.pairing_planet.dto.recipe;

import lombok.Builder;

import java.util.UUID;

@Builder
public record TrendingTreeDto(
        UUID rootRecipeId,
        String title,
        String culinaryLocale,
        Long variantCount,    // 변형 수
        Long logCount,        // 로그 수
        String latestChangeSummary // 최근 변형 사유 요약
) {}

File: lib/data/models/recipe/trending_tree_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'trending_tree_dto.g.dart';

@JsonSerializable()
class TrendingTreeDto {
  final String rootRecipeId;
  final String title;
  final String culinaryLocale;
  final int variantCount;
  final int logCount;
  final String? latestChangeSummary;

  TrendingTreeDto({
    required this.rootRecipeId,
    required this.title,
    required this.culinaryLocale,
    required this.variantCount,
    required this.logCount,
    this.latestChangeSummary,
  });

  factory TrendingTreeDto.fromJson(Map<String, dynamic> json) =>
      _$TrendingTreeDtoFromJson(json);
  Map<String, dynamic> toJson() => _$TrendingTreeDtoToJson(this);
}
